client-js/app/elements/labrad-plot.ts

```diff
@@ -562,11 +578,54 @@ export class Plot extends polymer.Base {
               .attr('y', (d) => p.yScale(p.yNext[d[1]]))
               .attr('width', (d) => p.xScale(p.xNext[d[0]]) - p.xScale(d[0]))
               .attr('height', (d) => Math.abs(p.yScale(p.yNext[d[1]]) - p.yScale(d[1])))
-              .style('fill', (d) => getColor(d[2], zMin, zMax));
+              .style('fill', (d) => getColor(d[p.displayTraces[0] + 2], zMin, zMax));
       }
       break;
     }
   }
+  
+  private selectTraces() {
+    this.$.traceSelector.open();
+  }
+
+  private selectAll() {
+    var checkboxes: any = document.getElementsByName('traces');
```

Comment:
Touching `document` breaks the encapsulation provided by polymer. You should use polymer's [dom api](https://www.polymer-project.org/1.0/docs/devguide/local-dom.html) instead, something like:
```typescript
var checkboxes = Polymer.dom(this.$.traceSelector).querySelectorAll('traces')
```